webapp/components/form/dropdown.js

```diff
@@ -63,25 +63,16 @@ class Dropdown extends React.Component {
   }
 
   async onSelectionChange(item) {
-    const { onBeforeChange, onChange, clearOnSelection, itemsLookupFunction } = this.props
-    const { items } = this.state
+    const { onBeforeChange, onChange, clearOnSelection } = this.props
 
-    let changed = false
     if (!onBeforeChange || (await onBeforeChange(item))) {
+      this.setState((statePrev) => ({
+        ...statePrev,
+        displayValue: clearOnSelection ? '' : this.getItemLabel(item),
+        opened: false,
+      }))
```

Comment:
I don't know if prevState is necessary here, it can be just
`this.setState({
  displayValue: clearOnSelection ? '' : this.getItemLabel(item),
  opened: false
})`

Reply:
that specific fix drove me crazy. The component was setting the state after on change; that means it was overriding props passed from outside after onChange.
it must be refactored anyway